actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -292,11 +292,18 @@ def to_hash
       if self.class.raise_on_unfiltered_parameters
         to_h.to_hash
       else
-        message = <<-DEPRECATE.squish
-          #to_hash unexpectedly ignores parameter filtering, and will change to enforce it in Rails 5.1.
-          Enable `raise_on_unfiltered_parameters` to respect parameter filtering, which is the default
-          in new applications. For the existing deprecated behaviour, call #to_unsafe_h instead.
-        DEPRECATE
+        if permitted?
+          message = <<-DEPRECATE.squish
+            `raise_on_unfiltered_parameters` is not enabled in your application.
+            Enable `raise_on_unfiltered_parameters` which is the default in new application.
+          DEPRECATE
+        else
+          message = <<-DEPRECATE.squish
+            #to_hash unexpectedly ignores parameter filtering, and will change to enforce it in Rails 5.1.
```

Comment:
Perhaps `#to_hash will raise on unpermitted parameters in Rails 5.1.` for this line instead.